modules/administration-guide/pages/default-extensions-for-microsoft-visual-studio-code.adoc

```diff
@@ -0,0 +1,137 @@
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      env:
+        - name: 'DEFAULT_EXTENSIONS'
+          value: '/projects/sample/extension.vsix'
+----
+
+* Use Devfile *postStart* event to fetch the extension binary from the network.
++
+It is possible to use *curl* or *wget* to download extensions to your workspace.
+For that you need to:
++
+** specify a devfile command to donload one or several extensions to your workpace
+** add a *postStart* event to run the command on workspace startup
+** define __DEFAULT_EXTENSIONS__ environment variable in the Devfile
++
+Following sample demonstrate what should be added to the devfile to add two extensions.
++
+[source,yaml]
+----
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      env:
+        - name: DEFAULT_EXTENSIONS
+          value: '/tmp/extension-1.vsix;/tmp/extension-2.vsix'
+
+commands:
+  - id: add-default-extensions
+    exec:
+      # name of the tooling container
+      component: tools
+      # download several extensions using curl
+      commandLine: |
+        curl https://.../extension-1.vsix --location -o /tmp/extension-1.vsix
+        curl https://.../extension-2.vsix --location -o /tmp/extension-2.vsix
+
+events:
+  postStart:
+    - add-default-extensions
+----
+
+* Include extension *.vsix* binaries to *che-code* image.
++
+Having default extensions bundled in the editor image along with __DEFAULT_EXTENSIONS__ environment variable defined in a config map, will allow you to use default extensions without the need to change the Devfile.
++
+Following steps will help you to add the extensions you need to the editor image.
++
+1. Create a directory, prepare and put one or several *.vsix* extensions to this directory.
++
+2. Create a Dockerfile with following content.
++
+[source,]
+----
+# inherit che-incubator/che-code:latest 
+FROM quay.io/che-incubator/che-code:latest
+USER 0
+
+# copy all .vsix files to /default-extensions directory
+RUN mkdir --mode=775 /default-extensions
+COPY --chmod=755 *.vsix /default-extensions/
+
+# add instruction to the script to copy default extensions to the working container
+RUN echo "cp -r /default-extensions /checode/" >> /entrypoint-init-container.sh
+----
++
+3. Build the image and then push it to the registry.
++
+[,console]
+----
+$ docker build -t yourname/che-code:next .
+$ docker push yourname/che-code:next
+----
++
+4. Add new config map, define __DEFAULT_EXTENSIONS__ environment variable, specify absolute paths to the extensions. 
++
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: vscode-default-extensions
+  labels:
+    controller.devfile.io/mount-to-devworkspace: 'true'
+    controller.devfile.io/watch-configmap: 'true'
+  annotations:
+    controller.devfile.io/mount-as: env
+data:
+  DEFAULT_EXTENSIONS: '/checode/default-extensions/extension1.vsix;/checode/default-extensions/extension2.vsix'
+----
```

Comment:
```suggestion
[source,yaml,subs="+quotes,+attributes,+macros"]
----
kind: ConfigMap
apiVersion: v1
metadata:
  name: vscode-default-extensions
  namespace: user1-{prod-short} <1>
  labels:
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-configmap: 'true'
  annotations:
    controller.devfile.io/mount-as: env
data:
  DEFAULT_EXTENSIONS: '/checode/default-extensions/extension1.vsix;/checode/default-extensions/extension2.vsix'
----
<1> An example user namespace.
```

Reply:
What if we leave it as it is to be able to create a config map by simple copy-paste?

Reply:
Sounds good to me, but in that case, could we add this suggestion? :

https://github.com/eclipse-che/che-docs/pull/2819#discussion_r1842347913

Reply:
Done